tests/plugins/test_deepspeed_plugin.py

```diff
@@ -830,3 +830,55 @@ def test_deepspeed_multigpu_no_schedulers(tmpdir):
     trainer.fit(model)
 
     _assert_save_model_is_equal(model, tmpdir, trainer)
+
+
+@RunIf(min_gpus=1, deepspeed=True, special=True)
+def test_deepspeed_warn_train_dataloader_called(tmpdir):
+    """
+    Test DeepSpeed warns when it calls ``train_dataloader`` internally for logging batch size.
+    """
+    model = BoringModel()
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        plugins=[DeepSpeedPlugin()],
+        gpus=1,
+        fast_dev_run=True,
+    )
+    with pytest.warns(UserWarning, match="Inferring the batch size for internal deepspeed logging"):
+        trainer.fit(model)
+
+
+@RunIf(min_gpus=1, deepspeed=True, special=True)
+def test_deepspeed_setup_train_dataloader(tmpdir):
+    """
+    Test DeepSpeed works when setup is required to call, and the user passes the batch size manually.
+    """
+
+    class PlDataModule(LightningDataModule):
+        def __init__(self):
+            super().__init__()
+            self._setup = False
+
+        def setup(self, stage: Optional[str] = None) -> None:
+            self._setup = True
```

Comment:
It should be:

```suggestion
            self._setup = torch.distributed.is_available() and torch.distributed.is_initialized()
```

Reply:
But this would fail for ddp_spawn. @awaelchli @ananthsub  I believe this is an important issue within Lightning right now.

Reply:
This doesn't apply to this module I think, this is only for DeepSpeed and Deepspeed doesnt support spawn!